core/eval.go

```diff
@@ -187,6 +187,19 @@ func evalSET(args []string) []byte {
 			if obj != nil {
 				return RESP_NIL
 			}
+		case "GET":
+			obj := Get(key)
+			tempObj := obj
+
+			if obj == nil {
+				return RESP_NIL
```

Comment:
IMO it is better to just update the response here. Let us invoke PUT at just one place.

Reply:
+1. We don't have to call PUT in 2 places. Passing GET will just change the return value of the SET command.

Reply:
Yes, actually, previously I did by using Put in the end only, but there was a review from someone which mentioned that the value won't get updated if I didn't add Put at 2 places, [this comment](https://github.com/DiceDB/dice/pull/246#discussion_r1704853008).

NVM, I'll make the changes 👍

Reply:
you don't need to add Put in two places but you will need a variable for a response and then you can return that in the end if it was of subtype GET instead of RESP_OK. If you just return it from here only then PUT never gets called and SET command doesn't execute. That is what I meant

Reply:
Hey! srivastav-yash, apologies for the miscommunication. Now I have got your point. But, actually, before doing that, I'm first trying to fix the issue with XX and NX options. If we're giving both xx and get, then the tests are failing.